src/state/controls/Button.ts

```diff
@@ -71,4 +71,11 @@ export class Button extends Control<IButtonData> implements IButton {
     public giveInput(input: IButtonInput): Promise<void> {
         return this.sendInput(input);
     }
+
+    public update(changedData: Partial<IButtonData>): Promise<void> {
+        if (changedData.cooldown) {
```

Comment:
IButtonData is documented as having its cooldown be a unix timestamp, here that is broken. Prefer to keep a single unambiguous representation for all data, providing utility methods as needed for fluent shortcuts.

Also, avoid mutating function input! This is a surprising side effect.